style_guide.md

```diff
@@ -0,0 +1,217 @@
+# Jsonnet Style Guide
+This style guide documents how we want to handle Jsonnet files when building the schemas with consistency. For more general principles and styling rules go to [Databricks repository](https://github.com/databricks/jsonnet-style-guide#databricks-jsonnet-guide).
+
+## Folder structure
+- The following structure is used:
+```
+england-wales/
+   ccs/
+      blocks/
+      lib/
+         rules.libsonnet
+   communal-establishment/
+   household/
+   individual/
+   lib/
+      common_rules.libsonnet
+      placeholders.libsonnet
+   ccs_household.jsonnet
+   census_communal_establishment.jsonnet
+   census_household.jsonnet
+   census_individual.jsonnet
+northern-ireland/
+
+```
+- England/Wales and Northern Ireland
+
+    - Jsonnet source files are separated into two folders: `england-wales` and `northern-ireland` with a number of sub-folders, each for corresponding schemas (e.g. `ccs`, `communal-establishment`, `household`, `individual`). These are divided into blocks in `blocks` folder.
+
+- Shared blocks
+
+    - Blocks from `individual` folder are used in both household and individual schema.
+
+- Top level Jsonnet files
+
+    - For each schema there are lists of block and placeholder imports.
+    - Schema structure is divided into sections, named after given schema and ending with `.jsonnet` suffix.
+
+- Lib folders
+
+    - Rules and placeholders are located in `lib` folder in corresponding schemas along with `blocks` folder.
+
+- Rules and placeholders
+
+    - These are separated into: `common_rules`, `placeholders` and `rules`. These files should end with `.libsonnet` suffix.
+
+- Top level and extended
+
+    - `common_rules` and `placeholders` are located in top-level "lib" folder.
+    - `rules` files for each schema are located in schema's sub-folder and extend some of the top-level `common_rules`.
```

Comment:
I think some of these points would be clearer if they were grouped together, something like:

```suggestion
- Jsonnet source files are separated into two top-level folders - `england-wales` and `northern-ireland` 
- The `lib/` folder contains common Jsonnet used across all questionnaire schemas
  - The file extension should be `.libsonnet`
  - Each file is grouped by function, currently `common_rules` and `placeholders`
- Each questionnaire has it's own folder, within this:
  - A top-level questionnaire definition Jsonnet file 
  - A `lib/` folder holds common Jsonnet used across multiple blocks
    - The file extension should be `.libsonnet`
    - Each file here can extend common Jsonnet from the questionnaire level lib folder
    - `rules.libsonnet` extends the top level `lib/common_rules.libsonnet`
  - A `blocks` folder:
    - Subdivided into section folders
    - Within each section folder a Jsonnet file per block
  
We currently have the following questionniares:

- `ccs`
- `communal-establishment`
- `household`
- `individual`

Blocks from the `individual` folder are also used from the `household` questionniare.
```